test/e2e/tests/tokens/nft/erc1155-interaction.spec.js

```diff
@@ -147,26 +135,20 @@ describe('ERC1155 NFTs testdapp interaction', function () {
         await driver.clickElement('#setApprovalForAllERC1155Button');
 
         // Wait for notification popup and check the displayed message
-        let windowHandles = await driver.waitUntilXWindowHandles(3);
-        await driver.switchToWindowWithTitle(
-          WINDOW_TITLES.Dialog,
-          windowHandles,
-        );
-        const displayedMessageTitle = await driver.findElement(
-          '[data-testid="confirm-approve-title"]',
-        );
-        assert.equal(
-          await displayedMessageTitle.getText(),
-          expectedMessageTitle,
-        );
-        const displayedUrl = await driver.findElement(
-          '.confirm-approve-content h6',
-        );
-        assert.equal(await displayedUrl.getText(), DAPP_URL);
-        const displayedDescription = await driver.findElement(
-          '.confirm-approve-content__description',
-        );
-        assert.equal(await displayedDescription.getText(), expectedDescription);
```

Comment:
all these above are anti-patterns

Reply:
Do you think we could write a regular expression (or a piece of code in general) to detect this anti-pattern?  Because then we could write a lint rule for it.

Reply:
whoa I think that is worth investigating, as this anti-pattern is re-introduced over and over in our tests unfortunately :( Despite having this documented as an anti-pattern. I can create a ticker for this.
Would it be something like: forbid anything in the form of ` assert.equal(await something.getText(), something)`

Reply:
Hmm you're thinking of a much simpler test than I was thinking of.  I was thinking of considering the `findElement` part too.  I think no matter what we write, there's going to be false positives and false negatives.  There are times when  `assert.equal(await something.getText(), something)` is actually the correct solution.  But I guess we can force an `eslint-disable-next-line` in those cases.

Reply:
:information_source:  as info, we have this ticket which reflects the work of removing this anti-pattern, and we've been adding some fixes around that:

https://github.com/MetaMask/metamask-extension/issues/19870

In parallel, adding this lint rule (if it's possible) could make that no more new tests are using this pattern anymore

Reply:
>  I was thinking of considering the findElement part too. I think no matter what we write, there's going to be false positives and false negatives. There are times when assert.equal(await something.getText(), something) is actually the correct solution
oh that is a really good point! Yeah maybe the rule needs to be a bit more elaborated indeed